nsv13/code/modules/overmap/ai-skynet.dm

```diff
@@ -120,7 +125,45 @@ Adding tasks is easy! Just define a datum for it.
 
 	return target
 
+/datum/fleet/proc/try_threat_elevation_reinforce()
+	if(!current_system)
+		return	//Which admin did this?
+	if(current_system.occupying_z)
+		return	//Loaded z - abort
+	if(current_system.check_conflict_status())
+		return //Bonking in progress - abort
+	if(world.time < last_encounter_time + TE_REINFORCEMENT_DELAY)
+		return
+	if(allow_difficulty_scaling)
+		//Account for pre-round spawned fleets.
+		if(SSovermap_mode?.mode)
+			applied_size = SSovermap_mode.mode.difficulty
+		else
+			applied_size = 1 //Lets assume a low number of players
+	else
+		applied_size = size
+	applied_size = CLAMP(applied_size, FLEET_DIFFICULTY_EASY, INFINITY)
+	if(threat_elevation_allowed)
+		applied_size += round(SSovermap_mode.threat_elevation / TE_POINTS_PER_FLEET_SIZE)	//Threat level modifies danger
+	if(destroyer_types?.len)
+		for(var/I=length(taskforces["destroyers"]); I<max(round(applied_size/2), 1);I++)
+			var/shipType = pick(destroyer_types)
+			var/obj/structure/overmap/member = new shipType()
+			add_ship(member, "destroyers")
+	if(battleship_types?.len)
+		for(var/I=length(taskforces["battleships"]); I<max(round(applied_size/4), 1);I++)
+			var/shipType = pick(battleship_types)
+			var/obj/structure/overmap/member = new shipType()
+			add_ship(member, "battleships")
+	if(supply_types?.len)
+		for(var/I=length(taskforces["supply"]); I<max(round(applied_size/4), 1);I++)
```

Comment:
Length() is null conditional
```suggestion
	if(length(destroyer_types))
		for(var/i = length(taskforces["destroyers"]); i < max(round(applied_size/2), 1); i++)
			var/shipType = pick(destroyer_types)
			var/obj/structure/overmap/member = new shipType()
			add_ship(member, "destroyers")
	if(length(battleship_types))
		for(var/i = length(taskforces["battleships"]); i < max(round(applied_size/4), 1); i++)
			var/shipType = pick(battleship_types)
			var/obj/structure/overmap/member = new shipType()
			add_ship(member, "battleships")
	if(length(supply_types))
		for(var/i = length(taskforces["supply"]); i < max(round(applied_size/4), 1); i++)
```

Reply:
Pretty sure all those are initialized as zero length lists regardless of whether there are ships in said group so this issue should not occur.